client/client.go

```diff
@@ -302,12 +302,28 @@ func addChange(cl *changelist.FileChangelist, c changelist.Change, roles ...stri
 	return nil
 }
 
-// AddDelegation creates a new changelist entry to add a delegation to the repository
-// when the changelist gets applied at publish time.  This does not do any validation
-// other than checking the name of the delegation to add - all that will happen
-// at publish time.
-func (r *NotaryRepository) AddDelegation(name string, threshold int,
-	delegationKeys []data.PublicKey, paths []string) error {
+// AddDelegation creates changelist entries to add provided delegation public keys and paths.
+// This method composes AddDelegationRoleAndKeys and AddDelegationPaths (each creates one changelist if called).
+func (r *NotaryRepository) AddDelegation(name string, delegationKeys []data.PublicKey, paths []string) error {
```

Comment:
I'm always for simplifying security critical code, and the majority of notary is pretty security critical.

If we don't see a legitimate use for thresholds, I say actually completely remove them. It's not that much extra code, but we should not ever justify carrying tech debt over some cloudy future promise.

I think we should make this as a unanimous team decision. If everyone agrees, we take it out.

/cc @endophage @NathanMcCauley